pkg/scanner/request.go

```diff
@@ -14,6 +14,13 @@ type Request struct {
 	ID string
 	// Thing to scan (e.g. URL, snippet of text, etc)
 	Resource resource.Resource
+	Errors   []RequestError
+}
+
+type RequestError struct {
+	Fatal   bool
+	Context string
+	Message string
 }
```

Comment:
What if we implement the [error interface](https://pkg.go.dev/builtin#error) and swap `Context` with a `Kind` field with strings like `CloneError`, `ResourceCleanupError`, etc?

Also instead of having a separate RequestError and ResponseError which require some mapping at the end, we could do a single `ScannerError` or even a more generic `LeakTKError` if we want to reuse this pattern in other bits of code.

Rationale:

- With a single error type we wouldn't need to do any mapping at the end
- We can change `Errors` type to `[]error` and tack what ever error type we want in it.
- We could (not in this MR) fix the multiple levels of escape characters issue with a custom error type by having it put them in a field on `LeakTKError` like `From []error` or something that we just collect the errors that bubbled up and caused this one. Then it'd render as just a nice clean JSON structure.
- If we implement the error interface we can start replacing other error creation in the code with this new error type without changing any of the code really. 
- Using `Kind` instead of `Context` helps communicate that this field is stable enough for things down the line to match on it and make decisions on it, and any human readable details would land in the Message field.

Thoughts?

Reply:
Pushed updated code. This should meet the above comment and our slack chat. Its extensible to do dots 2-4 from above at a later stage.